modules/schat.cpp

```diff
@@ -226,9 +230,15 @@ class CSChat : public CModule
 					Table.SetCell("Host", pSock->GetRemoteIP());
 					Table.SetCell("Port", CString(pSock->GetRemotePort()));
 					SSL_SESSION *pSession = pSock->GetSSLSession();
+#ifdef HAVE_OPAQUE_SSL_SESSION
+					if (pSession)
+						Table.SetCell("Cipher", SSL_CIPHER_get_name(SSL_SESSION_get0_cipher(pSession)));
+#else
 					if (pSession && pSession->cipher && pSession->cipher->name)
 						Table.SetCell("Cipher", pSession->cipher->name);
-
+#endif
+					else
+						Table.SetCell("Cipher", "None");
```

Comment:
Having looked at the implementation of `SSL_CIPHER_get_name` and digging through blame/history, it seems the function existed in the original SSLeay back in 1998 and has only had minor changes (adding consts to the signature) since then so my concern is no longer valid anyway. Additionally, it has always returned "(NONE)" when a NULL cipher is given, so slightly less code needed in schat.cpp.